<a href="https://colab.research.google.com/github/Huong1703/BT-Python/blob/main/Ex06_Housing_Market.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Housing Market

### Introduction:

This time we will create our own dataset with fictional numbers to describe a house market. As we are going to create random data don't try to reason of the numbers.

### Step 1. Import the necessary libraries

In [3]:
import pandas as pd
import random

### Step 2. Create 3 differents Series, each of length 100, as follows:
1. The first a random number from 1 to 4
2. The second a random number from 1 to 3
3. The third a random number from 10,000 to 30,000

In [7]:
random.seed(42)

series1 = pd.Series([random.randint(1, 4) for _ in range(100)])
series2 = pd.Series([random.randint(1, 3) for _ in range(100)])
series3 = pd.Series([random.randint(10000, 30000) for _ in range(100)])

print("Series 1:")
print(series1.head())

print("\nSeries 2:")
print(series2.head())

print("\nSeries 3:")
print(series3.head())

Series 1:
0    1
1    1
2    3
3    2
4    2
dtype: int64

Series 2:
0    2
1    3
2    2
3    3
4    2
dtype: int64

Series 3:
0    20827
1    12321
2    26847
3    17798
4    19125
dtype: int64


### Step 3. Let's create a DataFrame by joinning the Series by column

In [9]:
df = pd.DataFrame({'Column1': series1, 'Column2': series2, 'Column3': series3})

print("DataFrame:")
print(df)

DataFrame:
    Column1  Column2  Column3
0         1        2    20827
1         1        3    12321
2         3        2    26847
3         2        3    17798
4         2        2    19125
..      ...      ...      ...
95        2        3    29481
96        2        1    11302
97        4        1    12686
98        1        1    23737
99        4        1    29125

[100 rows x 3 columns]


### Step 4. Change the name of the columns to bedrs, bathrs, price_sqr_meter

In [11]:
df.rename(columns={'Column1': 'bedrs', 'Column2': 'bathrs', 'Column3': 'price_sqr_meter'}, inplace=True)

print("DataFrame with updated column names:")
print(df)

DataFrame with updated column names:
    bedrs  bathrs  price_sqr_meter
0       1       2            20827
1       1       3            12321
2       3       2            26847
3       2       3            17798
4       2       2            19125
..    ...     ...              ...
95      2       3            29481
96      2       1            11302
97      4       1            12686
98      1       1            23737
99      4       1            29125

[100 rows x 3 columns]


### Step 5. Create a one column DataFrame with the values of the 3 Series and assign it to 'bigcolumn'

In [14]:
bigcolumn = pd.concat([series1, series2, series3], axis=0)
df_bigcolumn = pd.DataFrame({'bigcolumn': bigcolumn})

print(df_bigcolumn)

    bigcolumn
0           1
1           1
2           3
3           2
4           2
..        ...
95      29481
96      11302
97      12686
98      23737
99      29125

[300 rows x 1 columns]


### Step 6. Oops, it seems it is going only until index 99. Is it true?

In [23]:
bigcolumn = pd.concat([series1, series2, series3], axis=0, ignore_index=True)
df_bigcolumn = pd.DataFrame({'bigcolumn': bigcolumn})

print(df_bigcolumn)


     bigcolumn
0            1
1            1
2            3
3            2
4            2
..         ...
295      29481
296      11302
297      12686
298      23737
299      29125

[300 rows x 1 columns]


### Step 7. Reindex the DataFrame so it goes from 0 to 299

In [25]:
df_bigcolumn = df_bigcolumn.reset_index(drop=True)

print(df_bigcolumn)

     bigcolumn
0            1
1            1
2            3
3            2
4            2
..         ...
295      29481
296      11302
297      12686
298      23737
299      29125

[300 rows x 1 columns]
